sdk/java-sdk-spring/src/main/java/kalix/javasdk/client/ComponentClient.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.javasdk.client;
+
+import kalix.spring.KalixClient;
+
+/**
+ * Utility to send requests to other Kalix components by composing a DeferredCall. To compose a call:
+ * 1. select component type (and pass id if necessary)
+ * 2. select component method, by Java method reference operator (::)
+ * 3. provide parameters (if required)
+ *
+ * <p>
+ * Example of use on a cross-component call:
+ * <pre>{@code
+ * public Effect<String> createUser(String userId, String email, String name) {
+ *   //validation here
+ *   var defCall = componentClient.forValueEntity(userId).call(UserEntity::createUser).params(email, name);
```

Comment:
Nice to have this example code. 💯 

We are using pure java, but that's not a common style. Good to have a clear example.